File: Prob3.c
# include <stdio.h>

long long numberToCheck = 600851475143;

int IsPrime(int number) {
    int i;
	
    for (i = 2; i < number; i++) {
        if (number % i == 0 && i != number){
			return 0;
		} 
    }
	
	if(numberToCheck % number == 0){
		printf("%d is a prime factor! \n", number);
		return 1;
	}
    return 1;
}

int main() {
	for(int i = 10000; i > 0; i--){
		IsPrime(i);
	}
	return 0;
}

File: Prob3.py
numberToCheck = 600851475143

def IsPrime(number):
	for x in range(2, number):
		if number % x == 0 and x != number:
			return 0

		if numberToCheck % number == 0:
			print number
			return 0

for x in xrange(10000,0, -1):
	IsPrime(x)